### PR TITLE
[nnc] Updated indices during broadcast to use int64_t

### PR DESCRIPTION
**Pull request by navahgar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64719
* #64716
* __->__ #64627

Differential Revision: [D30801686](https://our.internmc.facebook.com/intern/diff/D30801686)

### CODE REVIEW

**Comment by navahgar** [comment]
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30801686).

File: test/cpp/tensorexpr/test_kernel.cpp [hunk 1]
```diff
@@ -539,6 +539,40 @@ TEST_F(Kernel, CatInputTypesPromotion) {
...
+        %8 : Float(1, 1024, strides=[1024, 1], requires_grad=0, device=cpu) = aten::_cast_Float(%7, %2)
+        return (%8, %7))IR";
+
+  auto graph = std::make_shared<Graph>();
+  parseIR(graph_string, &*graph);
+
+  TensorExprKernel k(graph);
+  StmtPtr s = k.getCodeGenStmt();
+  std::ostringstream oss;
+  oss << *s;
```
Comment (ZolotukhinM) [inline test/cpp/tensorexpr/test_kernel.cpp:561]: This is unneeded since we're not checking the IR

**Review by ZolotukhinM — APPROVED** [review]
👍

**Comment by bertmaher** [comment]
It would be nice to have a python repro of this so it's not so tightly bound to the TE Kernel interface

**Review by bertmaher — APPROVED** [review]

**Review by bertmaher — CHANGES_REQUESTED** [review]
Hm, actually, lots of failures, seems like there's something still broken here.

File: test/cpp/tensorexpr/test_kernel.cpp (hunk 1 above)
Comment (navahgar) [reply test/cpp/tensorexpr/test_kernel.cpp]: Removed

**Comment by navahgar** [comment]
> It would be nice to have a python repro of this so it's not so tightly bound to the TE Kernel interface

Added

**Comment by navahgar** [comment]
Removed calling `simplify` before `inline`. This PR only includes the change to `LongImm` for indices during broadcast.

**Comment by navahgar** [comment]
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30801686).

**Pull request closed without merging**